flink-optimizer/src/main/java/org/apache/flink/optimizer/plantranslate/JobGraphGenerator.java

```diff
@@ -591,6 +605,9 @@ public void postVisit(PlanNode node) {
                                     .addOutputShipStrategy(ShipStrategyType.FORWARD);
                         }
                         chainedTask.setContainingVertex(container);
+                        if (!parallelismConfigured) {
```

Comment:
```suggestion
                        if (!node.isParallelismConfigured()) {
```

And remove the above `parallelismConfigured`.